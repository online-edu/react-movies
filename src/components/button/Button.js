import React from 'react';
import PropTypes from 'prop-types';

/**
 * Button component.
 */
const Button = ({ aria, tabIndex, label, click }) => (
    <button
        type="button"
        onClick={click}
        aria-label={label}
        tabIndex={tabIndex}
        {...aria}
        className="btn btn-primary my-2"
    >
        {label}
    </button>
);
/**
 * Button props types.
 */
Button.propTypes = {
    /** Id to be assigned to the control. */
    aria: PropTypes.object,
    /** Display value */
    label: PropTypes.string.isRequired,
    /** Gets called when the user clicks on button */
    click: PropTypes.func.isRequired,
    /** Maintain tab index for keyboard navigation. */
    tabIndex: PropTypes.number,
};
export default Button;
