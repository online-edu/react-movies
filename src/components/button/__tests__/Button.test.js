import React from 'react';
import { shallow } from 'enzyme';
import Button from '../Button';

describe('<Button />', () => {
    const onButtonClick = jest.fn();
    it('renders correctly', () => {
        const component = shallow(
            <Button label="Filter" click={onButtonClick} />,
        );
        expect(component).toMatchSnapshot();
    });

    it('contains class name "btn" ', () => {
        const wrapper = shallow(<Button label="Test" click={onButtonClick} />);
        expect(wrapper.find('.btn')).toHaveLength(1);
    });

    it('simulates click events', () => {
        const wrapper = shallow(<Button label="Test" click={onButtonClick} />);
        wrapper.find('button').simulate('click');
        expect(onButtonClick).toHaveBeenCalledTimes(1);
    });
});
