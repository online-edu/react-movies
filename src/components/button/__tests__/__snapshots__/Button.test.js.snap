// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    click={[MockFunction]}
    label="Filter"
    tabIndex={2}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-label": "Filter",
      "children": "Filter",
      "className": "btn btn-primary my-2",
      "onClick": [MockFunction],
      "tabIndex": 2,
      "type": "button",
    },
    "ref": null,
    "rendered": "Filter",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "Filter",
        "children": "Filter",
        "className": "btn btn-primary my-2",
        "onClick": [MockFunction],
        "tabIndex": 2,
        "type": "button",
      },
      "ref": null,
      "rendered": "Filter",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
